prime-router/src/main/kotlin/fhirengine/translation/hl7/HL7Truncator.kt

```diff
@@ -13,20 +14,22 @@ import kotlin.math.min
 /**
  * The shared HL7 truncation logic used in both the Covid Pipeline and the Universal Pipeline
  */
-class HL7Truncator {
+sealed interface HL7Truncator {
```

Comment:
Is there a reason we're splitting the truncators by pipeline instead of just adding the UP functionality? Because UP HL7 truncation should be able to handle the same truncation as the Covid pipeline, right?

Reply:
I split them out based on the types of parameters we were getting from each.

In the Covid pipeline we were getting an HL7 element like `PID-5-1` and then using a hardcoded map to find the path `/PATIENT_RESULT/PATIENT/PID` ([can see existing function here](https://github.com/CDCgov/prime-reportstream/blob/22c49438af032a57d8688427ac635cef29396571/prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/HL7Utils.kt#L101)).

For UP it directly hands us the full path `/PATIENT_RESULT/PATIENT/PID-5-1` so no need to look it up. This opens us up to use other message types as well by depending on the spec already handled the `Terser`.